participant/src/lib.rs

```diff
@@ -51,13 +60,94 @@ pub fn request_inputs(input: &mut impl BufRead, logger: &mut dyn Logger) -> Resu
 
     input.read_line(&mut signing_share_input).unwrap();
 
-    let signing_share =
-        <[u8; 32]>::from_hex(signing_share_input.trim()).map_err(|_| Error::MalformedSigningKey)?;
+    let signing_share = SigningShare::from_bytes(
+        <[u8; 32]>::from_hex(signing_share_input.trim()).map_err(|_| Error::MalformedSigningKey)?,
+    )
+    .unwrap(); // TODO: Need the correct error from frost
+
+    // TODO: Is extra validation needed here for public_key and signing_share or will that be resolved when used in generating key_packages etc.? Need to check
```

Comment:
Yep generating KeyPackage should validate the signature share.

Reply:
Removed comment [9c7d886](https://github.com/ZcashFoundation/frost-zcash-demo/pull/35/commits/9c7d886956b1599f46711d1b05d69059f0aacb1c)